internal/controller/etcdcluster_controller.go

```diff
@@ -99,7 +94,11 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if err := r.ensureClusterObjects(ctx, instance, isClusterReady); err != nil {
-		return ctrl.Result{}, fmt.Errorf("cannot create Cluster auxiliary objects: %w", err)
+		res, err2 := r.updateStatus(ctx, instance)
```

Comment:
Maybe name it as statusErr?

Reply:
Or as another option you can create function updateStatusOnErr and just always return this function in case of error. This function will wrap error and update status internally